app/gui/src/dashboard/hooks/projectHooks.ts

```diff
@@ -397,6 +398,40 @@ export function useRenameProjectMutation() {
 
       return backend.updateProject(id, { projectName: newName }, title)
     },
+    onMutate: async ({ newName, project }) => {
+      const queryKey = createGetProjectDetailsQuery.getQueryKey(project.id)
+      await client.cancelQueries({
+        queryKey,
+      })
+      // Optimistically update the project name.
+      const previousProjectDetails = client.getQueryData<backendModule.Project>(queryKey)
+      if (previousProjectDetails) {
+        client.setQueryData<backendModule.Project>(queryKey, {
+          ...previousProjectDetails,
+          name: newName,
+        })
+      }
+
+      return { queryKey }
+    },
+    onError: (_err, _variables, context) => {
+      if (context?.queryKey) {
+        const toInvalidate = [
+          {
+            predicate: (query: reactQuery.Query) =>
+              reactQuery.matchQuery({ queryKey: ['listDirectory'] }, query),
+          },
```

Comment:
Isn't constructing a predicate that calls `matchQuery` with a filter equivalent to passing the filter directly? (And likewise below)

```suggestion
          {
            queryKey: ['listDirectory'],
          },
```

Reply:
Yep, you are right, I confused it.